### ISSUE TITLE
Uncaught exception in Sound.seek()

### ISSUE THREAD

**Issue by EyeOfThReindeer** [goldfire/howler.js#886]
Hello,
I found a small, yet annoying bug, persisting in latest version. 

Every time, when `Sound.seek()` function is called with a seek parameter exceeding the sound's duration, Howler crashes with `Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause().`

I noticed that behaviour when I tried to create a progress bar as an active control, but accidentally multiplied the duration with full percents, instead of decimal value.

It seems to be related to another issue experienced on Chrome, when rapid pause/play calling resulted in the same exception. I believe it could be fixed with a single condition in `Sound.seek()`, comparing the seek parameter to sound's duration. 

I experienced the issue on Chrome 63.

Hope this helps. Regards,
- Jan

**Comment by goldfire** [comment]
I'm unable to get this to happen in 2.0.7 and the play/pause issue you mention was fixed in 2.0.6. I'll reopen if you are still seeing this on the latest howler.

**Comment by EyeOfThReindeer** [comment]
I avoided the issue by fixing my own mistake with the percent value representation, and did not investigate any further. 

If you couldn't recreate the error, it may have been something different. If you care to see my code, there it is (it is based on your playlist demo):

```
// CREATE AUDIO PLAYER CLASS
var Player = function (id, file) {
    this.init(id, file);
};

Player.prototype = {
    init: function (id, file) {
        var self = this;
        self.id = id;
        self.file = file;

        // INIT CONTROLS
        self.controls = {
            container: $("." + id),
            play: $("." + id + " .audio_widget__play"),
            pause: $("." + id + " .audio_widget__pause"),
            timer: $("." + id + " .audio_widget__timer"),
            duration: $("." + id + " .audio_widget__duration"),
            progress: $("." + id + " .audio_widget__progress"),
            progressBar: $("." + id + " .audio_widget__progressBar"),
        };
        self.controls.container.attr("data-listened", 0);
        // play / pause button
        $(self.controls.play).on('click', function () {
            self.play();
        });
        $(self.controls.pause).on('click', function () {
            self.pause();
        });
        // progressbar 
        $(self.controls.progressBar).on('click', function (e) {
            var perc = e.offsetX / $(this).width();
            if (perc > 1)
                perc = 1;
            self.seek(perc);
        });


        // LOAD THE TRACK and show its duration
        if (self.howl) {
        } else {
            self.howl = new Howl({
                src: [file],
                html5: true, // Force to HTML5 so that the audio can stream in (best for large files).
                onplay: function () {
                    requestAnimationFrame(self.step.bind(self));
                },
                onload: function () {// Display the duration.
                    self.controls.duration.html(
                            self.formatTime(Math.round(self.howl.duration()))
                            );
                },
                onend: function () {
                },
                onpause: function () {
                },
                onstop: function () {
                }
            });
        }


        self.controls.container.removeClass("playing");
    },
    /**
     * Play a song in the playlist.
     * @param  {Number} index Index of the song in the playlist (leave empty to play the first or current).
     */
    play: function () {
        var self = this;
        self.stopSiblings();
        self.howl.play();
        self.controls.container.addClass("playing");
        self.controls.container.trigger("audioPlayed");
    },
    /**
     * Stop all other present players
     */
    stopSiblings: function () {
        var self = this;
        $.each(window.audioPlayers, function (index, player) {
            if (!empty(player)) {
                if (index != self.id) {
                    player.pause();
                }
            }
        });
    },
    /**
     * Pause the currently playing track.
     */
    pause: function () {
        var self = this;
        if (self.howl.playing()) {
            self.howl.pause();
            self.controls.container.removeClass("playing");
            self.controls.container.trigger("audioPaused");
        }
    },
    /**
     * Seek to a new position in the currently playing track.
     * @param  {Number} per Percentage through the song to skip.
     */
    seek: function (per) {
        var self = this;
        var position = self.howl.duration() * per;
        if (self.howl.playing()) {
            self.pause();
        }
        self.howl.seek(position);
        self.controls.container.attr("data-listened", per);
        self.play();
    },
    /**
     * The step called within requestAnimationFrame to update the playback position.
     */
    step: function () {
        var self = this;
        // Determine our current seek position.
        var seek = self.howl.seek() || 0;
        self.controls.timer.html(
                self.formatTime(Math.round(seek))
                );
        var percents = (((seek / self.howl.duration()) * 100) || 0);
        self.controls.progress.css('width', percents + '%');

        self.controls.container.attr("data-listened", percents);
        self.controls.container.trigger("audioListened");

        // If the sound is still playing, continue stepping.
        if (self.howl.playing()) {
            requestAnimationFrame(self.step.bind(self));
        }
    },
    /**
     * Format the time from seconds to M:SS.
     * @param  {Number} secs Seconds to format.
     * @return {String}      Formatted time.
     */
    formatTime: function (secs) {
        var minutes = Math.floor(secs / 60) || 0;
        var seconds = (secs - minutes * 60) || 0;

        return minutes + ':' + (seconds < 10 ? '0' : '') + seconds;
    }
};


// INITIALIZE ALL AUDIO PLAYERS
function initPlayers() {
    $(".audio_widget__item").each(function (index, object) {
        var id = $(this).attr("data-id");
        var src = $(this).attr("data-src");
        window.audioPlayers[id] = new Player(id, src);
    });
}
$(window).load(function () {
    initPlayers();
});

// helper function
var empty = function (variable) {
    return (typeof (variable) == "undefined" || variable == null);
}
```

HTML markup:
```
<div class="audio_widget__list">
    <div class="audio_widget__item audio_widget__item_8" 
         data-id="audio_widget__item_8"
         data-src="XXX">
        <div class="audio_widget__content">
            <div class="audio_widget__controls">
                <div class="btn audio_widget__play">Play</div>
                <div class="btn audio_widget__pause">Pause</div>
            </div>

            <div class="audio_widget__time">
                <div class="audio_widget__timer">0:00</div>
                <div class="audio_widget__duration">0:00</div>
            </div>

            <div class="audio_widget__title">
                XXXX Title
            </div>

            <div class="audio_widget__links">
                <div class="audio_widget__links__inner">
                    <a href="XXXX" target="blank">
                        XXXX Link
                    </a>
                </div>
            </div>
        </div>
        <div class="audio_widget__progressBar">
            <div class="audio_widget__progress"></div>
        </div>
    </div>
    <!-- More simillar children -->
</div>
```
